### PR TITLE
regex matches now have an index associated with them

### PR DESCRIPTION
**Pull request by ermay12**

### CODE REVIEW

File: src/main/java/com/github/ermay12/regex/Regex.java [hunk 1]
```diff
@@ -669,13 +702,14 @@ public static Regex concatenate(Regex... components) {
...
+   *
    * @param ss the CharSequences that are the options
    * @return a regex which matches one of the given regular expressions
    */
   public static Regex oneOf(CharSequence... ss) {
-    assert(ss.length > 0); // Should not fail because you cannot call
-                           // oneOf with no arguments due to overloading
-    if(ss.length > 1) {
+    assert (ss.length > 0); // Should not fail because you cannot call
+    // oneOf with no arguments due to overloading
```
Comment (alex-stanescu) [inline src/main/java/com/github/ermay12/regex/Regex.java]: what why did you delete this now the formatting is bad

File: src/main/java/com/github/ermay12/regex/Regex.java [hunk 2]
```diff
@@ -693,13 +727,14 @@ public static Regex oneOf(CharSequence... ss) {
...
+   *
    * @param rs the regular expressions that are the options
    * @return a regex which matches one of the given regular expressions
    */
   public static Regex oneOf(Regex... rs) {
-    assert(rs.length > 0); // Should not fail because you cannot call
-                          // oneOf with no arguments due to overloading
-    if(rs.length > 1) {
+    assert (rs.length > 0); // Should not fail because you cannot call
+    // oneOf with no arguments due to overloading
```
Comment (alex-stanescu) [inline src/main/java/com/github/ermay12/regex/Regex.java]: ditto

File: src/main/java/com/github/ermay12/regex/Regex.java [hunk 3]
```diff
@@ -891,9 +939,17 @@ public String replace(CharSequence input, ReplacementRegex replacement) {
 
   public String replace(CharSequence input, Function<RegexMatch, String> l) {
     Matcher m = getMatcher(input);
-    return m.replaceAll(match -> l.apply(new RegexMatch(match, this)));
+    StringBuilder sb = new StringBuilder();
```
Comment (alex-stanescu) [inline src/main/java/com/github/ermay12/regex/Regex.java:942]: Shouldnt you be able to do this by using the stream that we have?

**Review by alex-stanescu — CHANGES_REQUESTED** [review]
Some small changes that need to be fixed

File: src/main/java/com/github/ermay12/regex/Regex.java (hunk 1 above)
Comment (ermay12) [reply src/main/java/com/github/ermay12/regex/Regex.java]: whoops auto-linter

File: src/main/java/com/github/ermay12/regex/Regex.java (hunk 2 above)
Comment (ermay12) [reply src/main/java/com/github/ermay12/regex/Regex.java]: ditto

File: src/main/java/com/github/ermay12/regex/Regex.java (hunk 3 above)
Comment (ermay12) [reply src/main/java/com/github/ermay12/regex/Regex.java:942]: Something about needing the matcher and not just a match result i think

File: src/main/java/com/github/ermay12/regex/Regex.java (hunk 1 above)
Comment (alex-stanescu) [reply src/main/java/com/github/ermay12/regex/Regex.java]: Can you un-auto-lint

File: src/main/java/com/github/ermay12/regex/Regex.java (hunk 3 above)
Comment (alex-stanescu) [reply src/main/java/com/github/ermay12/regex/Regex.java:942]: fine

**Review by alex-stanescu — APPROVED** [review]
